go/pkg/client/cas.go

```diff
@@ -938,40 +961,53 @@ func (c *Client) readBlobStreamed(ctx context.Context, d digest.Digest, offset,
 	if limit > 0 && limit < sz {
 		sz = limit
 	}
+	if err = w.Reset(); err != nil {
```

Comment:
Wait, if you moved this Reset outside of the closure, that tells me you no longer need it to be resettable. And you're removing the functionality of retrying on failed digest verification, too.
So, suggestion: base this PR on a revert of my https://github.com/bazelbuild/remote-apis-sdks/pull/271. Should be much simpler, imo.

Reply:
We'll have to clean this up in the fixit next week, then. There's a lot of stuff here now that no longer should be that will just be confusing. We no longer need the whole ResettableWriter struct and interface, and the refactoring I did that made everything more complicated...
But I'm okay with leaving it as a cleanup task for the next CL.